typing/env.ml

```diff
@@ -426,11 +428,15 @@ let read_pers_struct check modname filename =
            } in
   if ps.ps_name <> modname then
     error (Illegal_renaming(modname, ps.ps_name, filename));
+
   List.iter
     (function
         | Rectypes ->
             if not !Clflags.recursive_types then
               error (Need_recursive_types(ps.ps_name, !current_unit))
+        | Unsafe_string ->
+            if Config.safe_string then
+              error (Depend_on_unsafe_string_unit (ps.ps_name, !current_unit));
```

Comment:
The error is only raised when the system is *configured* with safe-string (this is `Config` above, not `Clflags`).  So no, no risk (unless I somehow screwed it up).